R-package/R/utils.R

```diff
@@ -103,18 +103,13 @@ xgb.Booster.check <- function(bst, saveraw = TRUE)
 ## ----the following are low level iteratively function, not needed if
 ## you do not want to use them ---------------------------------------
 # get dmatrix from data, label
-xgb.get.DMatrix <- function(data, label = NULL, missing = NULL, weight = NULL) {
+xgb.get.DMatrix <- function(data, label = NULL, missing = NA, weight = NULL) {
   inClass <- class(data)
   if (inClass == "dgCMatrix" || inClass == "matrix") {
     if (is.null(label)) {
       stop("xgboost: need label when data is a matrix")
     }
-    # dtrain <- xgb.DMatrix(data, label = label)
-    if (is.null(missing)){
-      dtrain <- xgb.DMatrix(data, label = label)
-    } else {
-      dtrain <- xgb.DMatrix(data, label = label, missing = missing)
-    }
+    dtrain <- xgb.DMatrix(data, label = label, missing = missing)
```

Comment:
I'm pretty sure the if(is.null) {} else {} part is no longer necessary since missing just defaults to NA everywhere now.  Let me know if I should change this though.